src/AddressSpace.cc

```diff
@@ -818,6 +818,14 @@ KernelMapping AddressSpace::map(Task* t, remote_ptr<void> addr,
     return m;
   }
 
+#if defined(__x86_64__)
+  if (t->session().is_recording() && t->arch() == SupportedArch::x86_64 &&
+      m.intersects(LA57_RANGE)) {
+    static_cast<RecordSession*>(&t->session())->trace_writer()
+      .note_virtual_address_size(57);
+  }
+#endif
```

Comment:
Instead of doing this architecture-specific stuff and defining LA57_RANGE, we could just compute the number of bits needed for the end address (or end address - 1) and store the max of all those in the trace. I think that would be slightly simpler than this, and also give us Aarch64.

I think you can use `__builtin_clz` for this.

Reply:
I don't think this is any simpler but I did it.